R/plotreg.R

```diff
@@ -1,394 +1,372 @@
-      co.names <- co.names[-remove.rows]
-      lower.outer <- models[[i]]@ci.low
-      upper.outer <- models[[i]]@ci.up
-      lower.inner <- NULL
-      upper.inner <- NULL
-      lower.outer <- lower.outer[-remove.rows]
-      upper.outer <- upper.outer[-remove.rows]
-      
-      # sort according to reorder.coef argument
-      dataframe <- data.frame(co.names, co, lower.outer, upper.outer)
-      dataframe <- reorder(dataframe, reorder.coef)
-      co.names <- dataframe[, 1]
-      co <- dataframe[, 2]
-      lower.outer <- dataframe[, 3]
-      upper.outer <- dataframe[, 4]
-      
-      signif.outer <- TRUE
-      note <- "Bars denote CIs."
-      message(paste0("Model ", i, ": bars denote  ", ci.level, 
-          " confidence intervals."))
+        lower.outer <- dataframe$ci.low
+        upper.outer <- dataframe$ci.up
+        lower.inner <- rep(0, length(dataframe$co.names))
+        upper.inner <- rep(0, length(dataframe$co.names))
+    }
+    
+    if (length(lower.outer) > 0) {
+        dataframe <- cbind(dataframe, lower.outer)     
+    }
+    if (length(lower.inner) > 0) {
+        dataframe <- cbind(dataframe, lower.inner)     
+    }
+    if (length(upper.outer) > 0) {
+        dataframe <- cbind(dataframe, upper.outer)     
+    }    
+    if (length(upper.inner) > 0) {
+        dataframe <- cbind(dataframe, upper.inner)     
     }
+    if (length(dataframe$pv) > 0) {
+        signif.outer <- dataframe$pv < (1 - ci.level)
+    } 
     
-    if (!is.null(custom.note)) {
-      note <- custom.note
+    # which terms are significant?
+    if(length(dataframe$ci.low) > 0) {
+        signif.outer <- ((dataframe$ci.low) > 0 & (dataframe$ci.up) > 0 | (dataframe$ci.low) < 0 & (dataframe$ci.up) < 0)
     }
     
+    if (class(signif.outer) == "logical" &&
+        length(signif.outer) == length(dataframe$co)) {
+        signif <- signif.outer == FALSE
+    } else if (signif.outer == TRUE) {
+        signif <- apply(cbind(lower.outer, upper.outer), 1,
+                        function(x) x[1] <= 0 && x[2] >= 0)
+    } else if (signif.outer == FALSE) {
+        signif <- apply(cbind(lower.inner, upper.inner), 1,
+                        function(x) x[1] <= 0 && x[2] >= 0)
+    } else {
+        stop("signif.outer does not correspond to the intervals provided.")
+    }
+    signif <- signif == FALSE
+    
+    
+    dataframe <- cbind(dataframe, signif)
+    
     if (length(co) == 0) {
-      stop(paste("No coefficients available. Was the 'omit.coef' argument", 
-          "misspecified? If coefficients were renamed using the",
-          "'custom.coef.names' argument, 'omit.coef' refers to the renamed",
-          "coefficients."))
+        stop(paste("No coefficients available. Was the 'omit.coef' argument", 
+                   "misspecified? If coefficients were renamed using the",
+                   "'custom.coef.names' argument, 'omit.coef' refers to the renamed",
+                   "coefficients."))
+    }
```

Comment:
Would it make sense to implement a `custom.coef.map` argument like in `screenreg` and other functions? For maximal compatibility?

Reply:
Have you added the `custom.coef.map` argument already?

Reply:
yes! :)

Reply:
Great! :)